lib/views/create_user/create_user_controller.ex

```diff
@@ -55,11 +58,22 @@ defmodule Bonfire.UI.Me.CreateUserController do
     end
   end
 
-  defp greet(conn, params, id, name) do
+  defp build_greet_message(name, is_admin) do
```

Comment:
can be simplified as (note also the `l` function for localisation)
```
  defp build_greet_message(name, is_admin) do
    flash_message = l("Hey %{name}, nice to meet you!", name: name)

      if is_admin do
        flash_message <> " " <> l("You have been promoted to admin!")
      else
        flash_message
      end
  end
```

or 
```
  defp build_greet_message(name, true) do
    build_greet_message(name, nil) <> " " <> l("You have been promoted to admin!")
  end
  defp build_greet_message(name, _) do
    l("Hey %{name}, nice to meet you!", name: name) 
  end

Reply:
That` if ` felt smelly but I wasn't sure what was the idiomatic way of doing it. I will update it